xwiki-platform-core/xwiki-platform-annotation/xwiki-platform-annotation-ui/src/main/resources/AnnotationCode/Script.xml

```diff
@@ -331,10 +331,14 @@ XWiki.Selection = Class.create({
     return descendant;
   },
 
+  // Removes the highligh on selected text (e.g., after pressing when annotating a new piece of text).
   removeSelectionHighlight : function() {
-    // unwrap
+    // We used previously replace for prototype, but it behaves badly when the parent of the wrapper is a td.
+    // Also, wrapper is always an element containing only text. In otherword, if &lt;strong&gt;a&lt;em&gt;b&lt;/em&gt;c&lt;/strong&gt; is
+    // selected for annotation, a, b, and c will each be wrapped in a &lt;span class='selection-highligh' ...&gt;.
+    // Therefore, when unwrapping we know that the spans of this.highlightWrappers contain only text.
```

Comment:
I think this comment about text nodes isn't needed anymore now that the code uses `childNodes` as they preserve everything including HTML elements.

Reply:
So you suggest leaving the comment unmodified? I find it a bit strange to highlight the fact that we only have text nodes several times even though the code would work perfectly fine if there were also other nodes. I'm not against having some comment, I have just the feeling it doesn't fully fit with the code below.

Reply:
No I'll remove the comment, but I'm lagging behind, I'll make the PR a draft until I clean that up.